src/core/tests/widgets/test_passwordinput.py

```diff
@@ -0,0 +1,54 @@
+import unittest
+from unittest.mock import MagicMock, Mock, patch
+import toga
+import toga_dummy
+
+
+class TestPasswordInput(unittest.TestCase):
+    """
+    """
+    def setUp(self):
+        self.factory = MagicMock()
+        self.factory.PasswordInput = MagicMock(return_value=MagicMock())
+
+        self.value = 'Password'
+        self.placeholder = 'Placeholder'
+        self.readonly = False
+
+        self.password_input = toga.PasswordInput(factory=self.factory)
+
+    def test_factory_called(self):
+        self.factory.PasswordInput.assert_called_with(interface=self.password_input)
+
+    # def test_setting_readonly_invokes_impl_method(self):
```

Comment:
Yes, because `PasswordInput` inherits most of its functionality from `TextInput` and because we only mock the _impl `PasswordInput` these tests would fail. I should probably remove them, right?